x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -371,4 +313,52 @@ private IndicesPermissionGroupDefinition(
             }
         }
     }
+
+    static SimpleRole buildFromRoleDescriptor(
+        final RoleDescriptor roleDescriptor,
+        final FieldPermissionsCache fieldPermissionsCache,
+        final RestrictedIndices restrictedIndices
+    ) {
+        // TODO handle this when we introduce remote index privileges for built-in users and roles. That's the only production code
+        // using this builder
+        assert false == roleDescriptor.hasRemoteIndicesPrivileges();
+        Objects.requireNonNull(fieldPermissionsCache);
```

Comment:
`fieldPermissionsCache` was nulleable before, however it was only ever `null` in test code. Opting to make it not nulleable since test code should not impact production code. Passing an instantiated `FieldPermissionsCache` is the relevant tests does not incur a notable overhead.